circuits/src/lib.rs

```diff
@@ -13,8 +13,374 @@
+        // and then with the note-pk of the other note:
+        // `(enc_A_npk_out_1, enc_B_npk_out_1)
+        sender_enc::gadget(
+            composer,
+            self.sender_pk,
+            self.signatures,
+            [
+                self.output_notes_info[0].note_pk,
+                self.output_notes_info[1].note_pk,
+            ],
+            [
+                self.output_notes_info[0].sender_blinder,
+                self.output_notes_info[1].sender_blinder,
+            ],
+            self.output_notes_info[0].sender_enc,
+            self.output_notes_info[1].sender_enc,
+            payload_hash,
+        )?;
+
+        Ok(())
+    }
+}
+
+impl<const H: usize, const I: usize> Default for TxCircuit<H, I> {
+    fn default() -> Self {
+        let sk =
+            SecretKey::new(JubJubScalar::default(), JubJubScalar::default());
+
+        let mut tree = Tree::<(), H>::new();
+        let payload_hash = BlsScalar::default();
+
+        let mut input_notes_info = Vec::new();
+        let note = Note::empty();
+        let item = Item {
+            hash: note.hash(),
+            data: (),
+        };
+        tree.insert(*note.pos(), item);
+
+        for _ in 0..I {
+            let merkle_opening = tree.opening(*note.pos()).expect("Tree read.");
+            input_notes_info.push(InputNoteInfo {
+                merkle_opening,
+                note: note.clone(),
+                note_pk_p: JubJubAffine::default(),
+                value: 0u64,
+                value_blinder: JubJubScalar::default(),
+                nullifier: BlsScalar::default(),
+                signature: SignatureDouble::default(),
+            });
+        }
+
+        let output_note_info_0 = OutputNoteInfo {
+            value: 0,
+            value_commitment: JubJubAffine::default(),
+            value_blinder: JubJubScalar::default(),
+            note_pk: JubJubAffine::default(),
+            sender_enc: [(JubJubAffine::default(), JubJubAffine::default()); 2],
+            sender_blinder: [JubJubScalar::default(), JubJubScalar::default()],
+        };
+        let output_note_info_1 = output_note_info_0.clone();
+
+        let output_notes_info = [output_note_info_0, output_note_info_1];
+
+        let root = BlsScalar::default();
+        let deposit = u64::default();
+        let max_fee = u64::default();
+
+        let signatures =
+            (SchnorrSignature::default(), SchnorrSignature::default());
+
+        Self {
+            input_notes_info: input_notes_info.try_into().unwrap(),
+            output_notes_info,
+            payload_hash,
+            root,
+            deposit,
+            max_fee,
+            sender_pk: PublicKey::from(&sk),
+            signatures,
+        }
+    }
+}
+
+/// Struct holding all information needed by the transfer circuit regarding the
+/// transaction input-notes.
+#[derive(Debug, Clone)]
+pub struct InputNoteInfo<const H: usize> {
+    /// the merkle opening for the note
```

Comment:
Ts here, or make them lower case in the other struct

Reply:
will fix in #233